dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/storage/S3Store.java

```diff
@@ -0,0 +1,98 @@
+package com.salesforce.dockerfileimageupdate.storage;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.salesforce.dockerfileimageupdate.utils.DockerfileGitHubUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class S3Store implements ImageTagStore {
+    private static final Logger log = LoggerFactory.getLogger(S3Store.class);
+    private static final Character S3_FILE_KEY_PATH_DELIMITER = ';';
+    public static final String s3Prefix = "s3";
+    private final AmazonS3 s3;
+    private final String store;
+
+    public S3Store(AmazonS3 s3, String store) {
+        this.s3 = s3;
+        this.store = store;
+    }
+
+    /* The store link should be a bucket name on S3. */
+    public void updateStore(String img, String tag) throws IOException {
+        if (store == null) {
+            log.info("Image tag store cannot be null. Skipping store update...");
+            return;
+        }
+        log.info("Updating store: {} with image: {} tag: {}...", store, img, tag);
```

Comment:
Lets use `debug` instead of `info` as `info` is usually the default log level.

Reply:
I think in this case, we might want to see what is going on, no? Debug would be for something much more detailed that we don't want to have polluting out logs. I think this log line is more signal that noise, but feel free to convince me that is not the case.

Reply:
"I think in this case, we might want to see what is going on, no?"  Only if something is failing.  If it not we will never look at these logs.  If it is we can conditionally enable debug logging and re-run.  If we want this information on every run I would advise building a file containing information about what was touched and what failed and return that as the output of the command.  Sort of like our published artifacts files.  DFIU also runs faster if it prints less.